teps/0041-tekton-component-versioning.md

```diff
@@ -0,0 +1,304 @@
+                version:
+                  type: string
+                  description: version of the component
+  scope: Cluster
+  names:
+    plural: info
+    singular: info
+    kind: Info
+    categories:
+      - tekton
+      - tekton-pipelines
+```
+
+And a CR should be added during installation which has the details
+about the version of the component or any other info required.
+
+```yaml
+apiVersion: tekton.dev/v1
+kind: Info
+metadata:
+  name: pipeline-info
+spec:
+  version: v0.18.0
+```
+
+`ClusterRole` and `ClusterRoleBinding`
+
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: pipelines-info-view-cr
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules:
+  - apiGroups: ["tekton.dev"]
+    resources: ["info"]
+    verbs: ["get", "list", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: pipelines-info-crbinding
+subjects:
+  - kind: Group
+    name: system:authenticated
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: pipelines-info-view
+```
+
+#### Advantages
+
+1. The CRD can be clusterwide so user need not to know the installed pipeline namespace.
+
+#### Disadvantages
+
+1. Where to store the CRD?
+
+   - Pipelines, Triggers or somewhere else?
+   - In each component then how to handle the changes?
+     - (Solution) Each component will have to maintain their own CRD so that they
+       can provide their respective information if they want to.
+
+### 2. Creating a ConfigMap
+
+Tekton components should create their own ConfigMap which contains
+the version information.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: pipeline-version
+  namespace: tekton-pipelines
+data:
+  version: v0.19.0
+```
+
+`Role` and `RoleBinding`
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: info
+  namespace: tekton-pipelines
+rules:
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    resourceNames: ["pipelines-version"]
+    verbs: ["get", "describe"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: info
+  namespace: tekton-pipelines
+subjects:
+  - kind: Group
+    name: system:authenticated
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: info
+```
+
+#### Advantages
+
+1. Other users not having access to the namespace can still have access
+   to the particular ConfigMap.
+
+   `kubectl get cm pipelines-version -n tekton-pipelines`
+
+#### Disadvantages
+
+1. User need to know the namespace in which Tekton components are installed
+   since the scope is only the namespace.
+
+## Test Plan
+
+Basic test scenarios of getting the version information should be added in all the components.
+Other components like CLI can add the test cases based on their usage.
+
+## Design Evaluation
```

Comment:
is this something we should be collaborating with the Operator folks about as well? im assuming we'd want to have 1 consistent way to handle this information

Reply:
@bobcatfish of course, but I wonder what do you mean about this ? Operator folks are reviewer here as well, so they are/should be collaborating already 🙃 
cc @tektoncd/experimental-maintainers

Reply:
> Operator folks are reviewer here as well

maybe this comes back to [our github discussion about blunderbuss](https://github.com/tektoncd/community/discussions/362), but [the way our TEP process has been defined so far](https://github.com/tektoncd/community/tree/main/teps#tep-review-slos) we're only using the assignees to gate proposals, i.e. given this:

![image](https://user-images.githubusercontent.com/432502/115058380-33c7d180-9eb3-11eb-8226-cd1b32b3bea7.png)

I'd expect @afrittoli @ImJasonH and me to review this and give an approve or lgtm, there might be operator folks in the reviewers but b/c reviewers are automatically assigned i wouldnt have taken that to be the list of ppl who need to ok this proposal before we accept it

Reply:
oh boy I pinged the wrong alias even 🤦🏼 


> I'd expect @afrittoli @ImJasonH and me to review this and give an approve or lgtm, there might be operator folks in the reviewers but b/c reviewers are automatically assigned i wouldnt have taken that to be the list of ppl who need to ok this proposal before we accept it

hum, yeah I don't mean operator folks need to be the one to validate this PR, just stating the fact that they are reviewers/owners of TEPs as well as other tektoncd components maintainer 😉